### PR TITLE
Feature/filter for test patients

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/185649710

see [related PR](https://github.com/uwcirg/isacc-messaging-client-sof/pull/63)

- A new environment variable, `ENABLE_FILTER_FOR_TEST_PATIENTS`, is used to determine whether to show the checkbox for test patients.
- A new environment variable, `FILTER_FOR_TEST_PATIENTS_LABEL`, is used to label the checkbox for filtering for test patients, e.g. `FILTER_FOR_TEST_PATIENTS_LABEL=Include test recipients` for ISACC
- Test patients will be hidden by default
- When the checkbox is checked, test patients will be included in the patient list

Screenshot of the UI
Screen Shot 2023-08-02 at 3 53 36 PM

### CODE REVIEW

**Review by mcjustin — APPROVED** [review]
Thanks for the quick work @achen2401 ! Perhaps we should switch to this branch for testing on dev...

**Review by mcjustin — CHANGES_REQUESTED** [review]
@achen2401  now that I see this live on the dev system, I realize that I didn't spec it out very well at https://www.pivotaltracker.com/story/show/185649710 .
Can you please make these changes?
1. Change the checkbox label "Test recipients" to "Include test recipients"
2. When the checkbox is checked, it should include all recipients, whether they are test or not.
3. Move the above checkbox & label to be underneath the "Recipients I am following" checkbox (low priority, ok to skip if it's complicated)

**Comment by achen2401** [comment]
> Can you please make these changes?
> 
> 1. Change the checkbox label "Test recipients" to "Include test recipients"
> 2. When the checkbox is checked, it should include all recipients, whether they are test or not.
> 3. Move the above checkbox & label to be underneath the "Recipients I am following" checkbox (low priority, ok to skip if it's complicated)

@mcjustin 2, 3 are now on shared dev instance.  For 1.  we just need to change the environment var, `FILTER_FOR_TEST_PATIENTS_LABEL` to say `Include test patients`

**Comment by mcjustin** [comment]
> @mcjustin 2, 3 are now on shared dev instance. For 1. we just need to change the environment var, `FILTER_FOR_TEST_PATIENTS_LABEL` to say `Include test patients`

@achen2401  this looks great to me now (ack the label still needs to be configured), merge when ready!

**Review by mcjustin — APPROVED** [review]

File: patientsearch/config.py [hunk 1]
```diff
@@ -25,7 +25,12 @@ def load_json_config(potential_json_string):
 ENABLE_PATIENT_DELETE = os.getenv("ENABLE_PATIENT_DELETE", "false").lower() == "true"
 
 ENABLE_PROVIDER_FILTER = os.getenv("ENABLE_PROVIDER_FILTER", "false").lower() == "true"
-
+FILTER_FOR_TEST_PATIENTS_LABEL = os.getenv(
+    "FILTER_FOR_TEST_PATIENTS_LABEL", "Include test patients"
```
Comment (pbugni) [inline patientsearch/config.py:29]: I updated just now on isacc-dev, but think this config is overkill, seems that we've settled on a good value and as we know, less config, is more ;)

File: patientsearch/src/js/components/PatientListTable.js [hunk 2]
```diff
@@ -675,6 +680,11 @@ export default function PatientListTable() {
     if (patientIdsByCareTeamParticipant && patientIdsByCareTeamParticipant.length) {
       apiURL += `&_id=${patientIdsByCareTeamParticipant.join(",")}`;
     }
+    if (getAppSettingByKey("ENABLE_FILTER_FOR_TEST_PATIENTS")) {
+      if (!filterByTestPatients) {
+        apiURL += `&_security:not=HTEST`;
```
Comment (pbugni) [inline patientsearch/src/js/components/PatientListTable.js:685]: ah, so nice and clean, in the end :)

**Review by pbugni — APPROVED** [review]

**Pull request merged**